generator/codegen/linkfollower_extensions.go

```diff
@@ -0,0 +1,392 @@
+	// The "Execute" suffix is hardcoded in the template,
+	// see https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/go/api.mustache
+	return strcase.ToCamel(in) + "Execute"
+}
+
+func generateFollowFuncName(in string) string {
+	return "Follow" + strings.TrimSuffix(strings.TrimPrefix(strcase.ToCamel(in), "List"), "Execute") + "Link"
+}
+
+func getFuncNameReplacementMap(operationIDs mapset.Set[string]) map[string]string {
+	executeFuncList := Map(operationIDs.ToSlice(), generateExecuteFnName)
+	funcNameMap := make(map[string]string)
+	for _, fnName := range executeFuncList {
+		funcNameMap[fnName] = generateFollowFuncName(fnName)
+	}
+
+	return funcNameMap
+}
+
+func getOperationIDs(swagger *openapi3.T) (opIDs mapset.Set[string], err error) {
+	opIDs = mapset.NewSet[string]()
+
+	for endpoint, pathInfo := range swagger.Paths {
+		var isCollection bool
+
+		// if no get field then not a collection
+		if pathInfo.Get != nil {
+			response200Val, subErr := getResponseFromPathInfo(pathInfo, endpoint, 200)
+			if subErr != nil {
+				err = subErr
+				return
+			}
+
+			var isRedacted bool
+			if c, ok := pathInfo.Get.ExtensionProps.Extensions[redactFlag].(json.RawMessage); ok {
+				err = json.Unmarshal(c, &isRedacted)
+				if err != nil {
+					return
+				}
+			}
+			if isRedacted {
+				continue
+			}
+
+			// if get not "application/json" then it is "application/octet-stream" and collections don't have that
+			if appJSON := response200Val.Content.Get(jsonMIME); appJSON != nil {
+				schemaVal, collectionRef, subErr := getCollectionSchema(swagger, appJSON, endpoint)
+				if subErr != nil {
+					err = subErr
+					return
+				}
+
+				if c, ok := schemaVal.ExtensionProps.Extensions[collectionFlag].(json.RawMessage); ok {
+					err = json.Unmarshal(c, &isCollection)
+					if err != nil {
+						return
+					}
+
+					if !isCollection || shouldIgnoreCollection(collectionRef) {
+						continue
+					}
+
+					itemRef, subErr := extractItemRef(schemaVal, collectionRef)
+					if subErr != nil {
+						err = subErr
+						return
+					}
+
+					if shouldIgnoreItem(itemRef) {
+						continue
+					}
+
+					opIDs.Add(pathInfo.Get.OperationID)
+				}
+
+				if isMessagesCollection := strings.HasSuffix(endpoint, "messages"); isMessagesCollection {
+					if strings.Contains(collectionRef, notificationFeedRef) || strings.Contains(collectionRef, messageItemRef) {
+						if shouldIgnoreItem(collectionRef) {
+							continue
+						}
+
+						opIDs.Add(pathInfo.Get.OperationID)
+					}
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func renderSrc(fn *ast.FuncDecl, fset *token.FileSet) (src string, err error) {
+	var buf bytes.Buffer
+	renderErr := printer.Fprint(&buf, fset, fn)
+	if renderErr != nil {
+		err = commonerrors.Newf(commonerrors.ErrUnexpected, "Could not render source code for func '%s()'", fn.Name.Name)
```

Comment:
leave out the `()`, it isn't guaranteed that a function has no args so just use its name